stackit/internal/services/mongodbflex/instance/resource.go

```diff
@@ -294,8 +288,9 @@ func (r *instanceResource) Create(ctx context.Context, req resource.CreateReques
 		if resp.Diagnostics.HasError() {
 			return
 		}
-		r.loadFlavorId(ctx, &resp.Diagnostics, &model, flavor)
-		if resp.Diagnostics.HasError() {
+		err := loadFlavorId(ctx, r.client, &model, flavor)
+		if err != nil {
+			core.LogAndAddError(ctx, &resp.Diagnostics, "Error loading flavor ID", fmt.Sprintf("getting flavor ID from CPU and RAM specifications: %v", err))
```

Comment:
Summary should be `Error updating instance`, to be harmonized with other logs, and detail should be capitalized (and could also be simplified, maybe to `Loading flavor: %v` or something of the sort)